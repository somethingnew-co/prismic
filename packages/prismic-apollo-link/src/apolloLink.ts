import Prismic from 'prismic-javascript'
import { ApolloLink, HttpLink } from '@apollo/client'
import { setContext } from '@apollo/client/link/context'

interface Config {
  repositoryName: string
  accessToken?: string
  defaultLocale?: string
}

interface PreviewData {
  [prop: string]: any
}

interface PrismicApolloLink {
  (config: Config, previewData?: PreviewData): ApolloLink
}

const prismicApolloLink: PrismicApolloLink = (config, previewData) => {
  const { repositoryName, accessToken, defaultLocale } = config
  const apiEndpoint = `https://${repositoryName}.cdn.prismic.io/api/v2`
  const graphqlEndpoint = `https://${repositoryName}.prismic.io/graphql`

  const prismicLink = setContext(async (req, previousContext) => {
    const prismicClient = Prismic.client(apiEndpoint, { accessToken })
    const prismicApi = await prismicClient.getApi()

    const headers = {
      ...previousContext.headers,
      'Accept-Language': defaultLocale || 'en-us',
      'Prismic-ref': previewData?.ref || prismicApi.masterRef.ref,
    }

    if (accessToken) {
      headers.Authorization = `Token ${accessToken}`
    }

    return { headers }
  })

  const httpLink = new HttpLink({
    fetch: (url, options) => fetch(url, options),
    uri: graphqlEndpoint,
    useGETForQueries: true,
  })

  return ApolloLink.from([prismicLink, httpLink])
}

export default prismicApolloLink
