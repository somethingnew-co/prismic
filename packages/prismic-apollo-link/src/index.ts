export { default as prismicApolloLink } from './apolloLink'
