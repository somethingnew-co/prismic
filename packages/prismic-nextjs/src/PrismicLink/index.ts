export { default as PrismicLink } from './PrismicLink'
