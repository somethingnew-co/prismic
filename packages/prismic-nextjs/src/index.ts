export { Link } from './Link'
export * from '@stnew/prismic-react'
