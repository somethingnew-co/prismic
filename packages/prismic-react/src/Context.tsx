import { createContext } from 'react'
import { Context } from './types'

const prismicContext: Context = {
  slices: {},
  linkResolver: () => '/',
  hrefResolver: () => '/',
  rootResolver: '',
}

const PrismicContext = createContext(prismicContext)

export default PrismicContext
