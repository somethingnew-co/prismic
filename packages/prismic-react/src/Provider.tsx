import React from 'react'
import PrismicContext from './Context'
import { Provider } from './types'

function PrismicProvider({ slices, linkResolver, hrefResolver, rootResolver, children }: Provider): JSX.Element {

  const value = {
    slices: Object.freeze(slices),
    linkResolver,
    hrefResolver,
    rootResolver,
  }

  return <PrismicContext.Provider value={value}>{children}</PrismicContext.Provider>
}

export default PrismicProvider
