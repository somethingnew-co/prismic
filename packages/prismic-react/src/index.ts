export { default as PrismicContext } from './Context'
export { default as PrismicProvider } from './Provider'
export { default as usePrismic } from './usePrismic'
export { default as useHtmlSerializer } from './useHtmlSerializer'
export { SliceZone } from './SliceZone'
export { default as serializeElements } from './serializeElements'
export * from '@stnew/prismic'
