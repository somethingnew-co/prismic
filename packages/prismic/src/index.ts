export { default as urlResolver } from './urlResolver'
