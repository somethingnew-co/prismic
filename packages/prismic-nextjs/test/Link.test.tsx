import React from 'react'
import { Link } from '../src'
import { renderWithPrismicProvider } from './test-setup'

const doc1 = {
  type: 'page',
  uid: 'homepage',
  link_type: 'Document',
  url: '',
}

const doc2 = {
  type: 'page',
  uid: 'other-page',
  link_type: 'Document',
  url: '',
}

const doc3 = {
  type: 'blog',
  uid: '',
  link_type: 'Document',
  url: '',
}

const doc4 = {
  type: 'post',
  uid: 'hello-world',
  link_type: 'Document',
  url: '',
}

const media = {
  link_type: 'Media',
  url: 'https://somethingnew.co/share.png',
}

const web = {
  link_type: 'Web',
  url: 'https://somethingnew.co',
}

const webEx = {
  link_type: 'Web',
  url: 'https://somethingnew.co',
  target: '_blank',
}

test('Prismic Doc', () => {
  const { container } = renderWithPrismicProvider(<>
    <Link href={doc1}>Document link</Link>
    <Link href={doc2}>Document link</Link>
    <Link href={doc3}>Document link</Link>
    <Link href={doc4}>Document link</Link>
    <Link href={media}>Media link</Link>
    <Link href={web}>Web link</Link>
    <Link href={webEx}>Web link (target blank)</Link>
    <Link href={webEx}>Web link (target blank)</Link>
  </>)
  expect(container).toMatchSnapshot()
})


test('String', () => {
  const { container } = renderWithPrismicProvider(<>
    <Link href="/">I am a link</Link>
    <Link href="/blog">I am a link</Link>
    <Link href="/test">I am a link</Link>
    <Link href="post" uid="hello-world">I am a link</Link>
    <Link href="https://somethingnew.co">I am a link</Link>
  </>)
  expect(container).toMatchSnapshot()
})
