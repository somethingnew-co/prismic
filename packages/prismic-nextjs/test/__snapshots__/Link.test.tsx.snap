// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prismic Doc 1`] = `
<div>
  <a
    href="/"
  >
    Document link
  </a>
  <a
    href="/other-page"
  >
    Document link
  </a>
  <a
    href="/blog"
  >
    Document link
  </a>
  <a
    href="/blog/hello-world"
  >
    Document link
  </a>
  <a
    href="https://somethingnew.co/share.png"
    rel="noopener noreferrer"
    target="_blank"
  >
    Media link
  </a>
  <a
    href="https://somethingnew.co"
  >
    Web link
  </a>
  <a
    href="https://somethingnew.co"
    rel="noopener noreferrer"
    target="_blank"
  >
    Web link (target blank)
  </a>
</div>
`;

exports[`String 1`] = `
<div>
  <a
    href="/"
  >
    I am a link
  </a>
  <a
    href="/blog"
  >
    I am a link
  </a>
  <a
    href="/test"
  >
    I am a link
  </a>
  <a
    href="/blog/hello-world"
  >
    I am a link
  </a>
  <a
    href="https://somethingnew.co"
    rel="noopener noreferrer"
    target="_blank"
  >
    I am a link
  </a>
</div>
`;
